erlang-ts.el

```diff
@@ -306,7 +422,8 @@ Use `treesit-font-lock-level' or `treesit-font-lock-feature-list'
   (advice-add #'erlang-font-lock-level-3 :around #'erlang-ts--font-lock-level-3)
   (advice-add #'erlang-font-lock-level-4 :around #'erlang-ts--font-lock-level-4)
 
-  (treesit-major-mode-setup))
+  (treesit-major-mode-setup)
+  (setq-local syntax-propertize-function #'erlang-ts--syntax-propertize))
```

Comment:
I don't think it's prudent to use `syntax-propertize-function` in a TreeSitter-powered mode, as you'll font-locking stuff at two different levels.

Reply:
it's used to modify char syntax in captured string.
since:
```
...
  :syntax-table erlang-mode-syntax-table
...
```

erlang-mode marked char `$` as `"/"`, aka `character-quote`.
see also https://github.com/erlang/otp/blob/master/lib/tools/emacs/erlang.el#L1569

set syntax-propertize-function only for put-text-property to avoid `'`, `"`, `$` are incorrectly highlighted in string or binary, and now there are triple-quoted string.

(Also, some emacs packages use char-syntax to jump between words.
The erlang char `$a` causes these packages to try to find another `$` character to match the bracket)

A new commit copied `erlang-mode-syntax-table` and unset `$`.